net-p2p/biglybt/biglybt-3.8.0.2.ebuild

```diff
@@ -0,0 +1,122 @@
+# sun.net.spi.nameservice.NameServiceDescriptor: Provider com.biglybt.core.util.spi.AENameServiceDescriptor not found
+#
+# Also it fails with >=jdk-23 so we limit the max version #936549
+DEPEND="${CP_DEPEND}
+	|| (
+		virtual/jdk:11
+		virtual/jdk:17
+		virtual/jdk:21
+	)
+"
+RDEPEND="${CP_DEPEND}
+	>=virtual/jre-1.8:*
+"
+
+DOCS=( README.md ChangeLog.txt )
+
+PATCHES=(
+	"${FILESDIR}/${PN}-3.2.0.0-disable-SWTUpdateChecker.patch"
+	"${FILESDIR}/${PN}-3.2.0.0-disable-shared-plugins.patch"
+	"${FILESDIR}/${PN}-3.5.0.0-Entities.javadoc.patch"
+	"${FILESDIR}/${PN}-3.6.0.0-disable-PluginUpdatePlugin.patch"
+	"${FILESDIR}/${PN}-3.6.0.0-fix-ambiguous.patch"
+)
+
+JAVA_GENTOO_CLASSPATH_EXTRA="target/classes"
+JAVA_JAR_FILENAME="BiglyBT.jar"
+JAVA_MAIN_CLASS="com.${PN}.ui.Main"
+JAVA_RESOURCE_DIRS=( {core,uis}/resources )	# yes, need them both
+JAVA_SRC_DIR="uis/src"
+
+declare -r ASSET_DIR="${S}"/assets/linux
+src_prepare() {
+	default #780585
+	java-pkg-2_src_prepare
+
+	# AENameServiceDescriptor fails to compile with jdk >= 11
+	# "error: package sun.net.spi.nameservice does not exist"
+	# https://github.com/BiglySoftware/BiglyBT/pull/2611
+	rm -r core/src/com/biglybt/core/util/spi || die
+
+	cp -r --reflink=always core/{src,resources} || die
+	rm -r core/resources/META-INF || die
+	find core/resources -type f -name "*.java" -delete || die
+	find core/resources -type f -name '.editorconfig' -delete || die
+
+	cp -r --reflink=always uis/{src,resources} || die
+	find uis/resources -type f -name "*.java" -delete || die
+
+	# patch he desktop file
+	sed -i '/^Exec=/d' "${ASSET_DIR}"/${PN}.desktop || die
+	echo "Exec=${PN}" >> "${ASSET_DIR}"/${PN}.desktop
+	sed -i '/^Icon=/d' "${ASSET_DIR}"/${PN}.desktop || die
+	echo "Icon=${PN}" >> "${ASSET_DIR}"/${PN}.desktop
+
+	# https://github.com/BiglySoftware/BiglyBT/pull/3523
+	sed -i 's/Application;//g' "${ASSET_DIR}"/${PN}.desktop || die
+}
+
+src_compile() {
+	# build core classes, needed for compiling uis
+	ejavac -d target/classes \
+		-cp "$(java-pkg_getjars commons-cli):$(java-pkg_getjars swt-${SWT_SLOT})" \
+		$(find core/src -name "*.java") || die
+
+	java-pkg-simple_src_compile
+
+	# for 3.8.0.2 it produces 2 errors:
+	# uis/src/com/biglybt/ui/swt/plugin/net/buddy/swt/BuddyPluginView.java:68:
+	# uis/src/com/biglybt/ui/swt/plugin/net/buddy/swt/BuddyPluginViewChat.java:45:
+	# error: package com.biglybt.ui.swt.plugin.net.buddy does not exist
+	# use doc && JAVA_SRC_DIR=( {core,uis}/src ) ejavadoc
+}
+
+src_install() {
+	java-pkg-simple_src_install
+
+	doicon -s 256 "${ASSET_DIR}"/${PN}.png
+	doicon -s scalable "${ASSET_DIR}"/${PN}.svg
+	domenu "${ASSET_DIR}"/${PN}.desktop
+
+	if use source; then
+		java-pkg_dosrc "core/src/*"
+		java-pkg_dosrc "uis/src/*"
+	fi
+	default
+}
+
+pkg_postinst() {
+	xdg_pkg_postinst
+
+	optfeature_header "If you are using plugin proxies you might want to swap them for these native packages:"
```

Comment:
I'm not really sure about the phrasing of this, but here's a grammar fix nonetheless:
```
suggestion
	optfeature_header "If you are using plugin proxies, you might want to swap them for these native packages:"
```